test_vae_training/0_explore_data.py

```diff
@@ -0,0 +1,62 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:microbe] *
+#     language: python
+#     name: conda-env-microbe-py
+# ---
+
+import pandas as pd
+import seaborn as sns
+import numpy as np
+import matplotlib.pyplot as plt
+from cm_modules import paths
+
+# +
+# Load dataset
+# Each column is a bacterial taxon, and each row is a single sample.
+# Rows are named using the format "${PROJECT_ID}_${SAMPLE_ID}"
+# Each cell is a read count.
+# There are ~11,800 samples in the preliminary dataset.
+# sample x taxon
+
+microbiome_data_filename = paths.RAW_MICROBIOME_DATA
+microbiome_data = pd.read_csv(microbiome_data_filename, sep="\t", index_col=0, header=0)
+# -
+
+print(microbiome_data.shape)
+microbiome_data.head()
+
+# How sparse is this data matrix?
+# How many 0s across the entire matrix?
+num_zeros = (microbiome_data == 0).sum().sum()
```

Comment:
You probably don't need to worry about this with your dataset size, but have you considered a sparse matrix format? Like single cell studies uses the `.mtx` file format, where instead of having a full matrix you list out the row and column ids and values for all nonzero cells. Could save space if your data is sparse enough.

Reply:
Super helpful comment! I was just using the data format that was provided to me. Since I'm used to working in matrices and the size isn't an issue I think I'll leave it for now. But very helpful to know